### PR TITLE
Strip disability names (and add test for it)

### PR DESCRIPTION
**Pull request by acovrig**
## Description of change

Trim spaces from the beginning/end of the disability names.

## Original issue(s)
[API-19014](https://vajira.max.gov/browse/API-19014)

### CODE REVIEW

**Comment by jw81** [comment]
Interesting.
We have existing logic that massages disability names prior to sending to EVSS over [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/models/claims_api/auto_established_claim.rb#L290).

Would you have any objections to moving your logic over to there, so we could keep that kind of logic in one place?

**Review by stiehlrod — DISMISSED** [review]

**Review by stiehlrod — COMMENTED** [review]
So for [API-19677](https://vajira.max.gov/browse/API-19677), there was a leading space AND a period at the end. Once both were removed, the claim was established. Do we want to adjust the regex with this ticket, or create a new one? @acovrig @jw81

**Comment by jw81** [comment]
> So for [API-19677](https://vajira.max.gov/browse/API-19677), there was a leading space AND a period at the end. Once both were removed, the claim was established. Do we want to adjust the regex with this ticket, or create a new one? @acovrig @jw81

Let's just let this PR focus on the whitespace for now.
We can create an additional ticket/PR if we need to for anything else.

File: modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb [hunk 1]
```diff
@@ -1327,6 +1327,20 @@ def obj.class
...
+        temp = JSON.parse(temp)
+        temp['data']['attributes']['autoCestPDFGenerationDisabled'] = auto_cest_pdf_generation_disabled
+        temp['data']['attributes']['claimDate'] = claim_date
+        temp['data']['attributes']['applicationExpirationDate'] = (Time.zone.today + 1.day).to_s
+        temp['data']['attributes']['disabilities'][0]['name'] =
+          " #{temp['data']['attributes']['disabilities'][0]['name']} "
+
+        temp.to_json
+      end
+    end
```
Comment (jw81) [inline modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb:1342]: I don't think ya need this.
It doesn't look like there's any spec logic that is making any expectations on that stripping behavior.

**Review by stiehlrod — APPROVED** [review]

**Review by jw81 — APPROVED** [review]

**Pull request merged**